services/harmony/app/frontends/ogc-edr/get-data-for-trajectory.ts

```diff
@@ -0,0 +1,280 @@
+
+  // Ensure the polygon is counter-clockwise
+  polygonPoints = ensureCounterClockwise(polygonPoints);
+
+  return polygonPoints;
+}
+
+/**
+* Converts a WKT LineString coords part into a list of polygons by splitting the LineString into
+* two-point segments and converting each segment into a polygon.
+*
+* @param wktLineCoords - The WKT LineString coords part in string to convert.
+* @param sideLength - The buffer distance (side length) to create around each segment.
+* @returns The WKT Polygon coords part in string with counter-clockwise ordered points.
+* @throws RequestValidationError - If the input does not have at least two points.
+*/
+function convertLineStringCoordsToPolygons(wktLineCoords: string, sideLength: number): string[] {
+  const points = wktLineCoords.split(',').map(coord => {
+    const [x, y] = coord.trim().split(/\s+/).map(Number);
+    return { x, y };
+  });
+
+  if (points.length < 2) {
+    throw new Error('LineString must contain at least two points');
+  }
+
+  const polygons: string[] = [];
+
+  // Loop through consecutive points to create line segments
+  for (let i = 0; i < points.length - 1; i++) {
+    const segment = [points[i], points[i + 1]];
+
+    // Convert each segment into a polygon
+    const polygonPoints = convertSegmentToPolygon(segment, sideLength);
+
+    // Convert polygon points to WKT format
+    const polygonString = polygonPoints.map(p => `${p.x} ${p.y}`).join(', ');
+
+    polygons.push(`((${polygonString}))`);
+  }
+
+  return polygons;
+}
+
+/**
+* Converts a WKT LINESTRING into a WKT POLYGON/MULTIPOLYGON by splitting the LINESTRING into
+* two-point segments and converting each segment into a polygon.
+*
+* @param wktLineString - The WKT LINESTRING to convert.
+* @param sideLength - The buffer distance (side length) to create around each segment.
+* @returns The WKT POLYGON/MULTIPOLYGON string with counter-clockwise ordered points for each polygon.
+* @throws RequestValidationError - If the input is not a valid WKT LINESTRING.
+*/
+function convertLineStringToPolygon(wktLineString: string, sideLength: number): string {
+  validateWkt(wktLineString);
+  const match = wktLineString.match(/LINESTRING\s*\((.*)\)/);
+  if (!match) {
+    throw new RequestValidationError(`query parameter "coords" invalid WKT LINESTRING format: ${wktLineString}`);
+  }
+
+  const polygons = convertLineStringCoordsToPolygons(match[1], sideLength);
+
+  let wktPolygon = '';
+  if (polygons.length > 1) {
+    wktPolygon = `MULTIPOLYGON (${polygons.join(', ')})`;
+  } else {
+    wktPolygon = `POLYGON ${polygons.join(', ')}`;
+  }
+
+  return wktPolygon;
+}
+
+/**
+ * Removes unnecessary spaces between LineStrings in a WKT MULTILINESTRING.
+ *
+ * @param wktMultiLineString - The WKT MULTILINESTRING to process.
+ * @returns The WKT MULTILINESTRING with the spaces removed.
+ */
+function cleanMultiLineString(wktMultiLineString: string): string {
```

Comment:
nit: Not sure it's worth having an entire function that just invokes a single regex.

Reply:
Dropped this function.